services/authorization-service.js

```diff
@@ -0,0 +1,86 @@
+const ERROR = require('../constants/error-constants');
+
+class AuthorizationService {
+
+    constructor(configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    /**
+     * Function to get a user's scopes for a given permission
+     * @param user - the current user
+     * @param permission - the name of the permission without scopes, please use a permissions constant for this input
+     * Example: "entity:action"
+     * @returns {*|string[]|*[]} - an array containing the scopes for the permission or an empty array, all scopes will be lowercase
+     */
+    getPermissionScope(user, permission){
```

Comment:
We also need to return scope value(s). For example, if scope is **study**, then we need to return a list of study ids associated with the user.

Reply:
I'll work on this change. I might have some questions but I will reach out if I need any help

Reply:
I've rewrote the getPermissionScope function to return {scopes: string[], scopeValues: string[]}